src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+            // Span<int> s = arr();
+            Diagnostic(ErrorCode.WRN_SameFullNameThisAggAgg, "Span<int>").WithArguments("", "System.Span<T>", "Span1, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null", "System.Span<T>").WithLocation(2, 1));
+
+        verifier.VerifyIL("<top-level-statements-entry-point>", """
+            {
+              // Code size       12 (0xc)
+              .maxstack  1
+              IL_0000:  call       "int[] Program.<<Main>$>g__arr|0_0()"
+              IL_0005:  newobj     "System.Span<int>..ctor(int[])"
+              IL_000a:  pop
+              IL_000b:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_SemanticModel()
+    {
+        var source = """
+            class C
+            {
+                System.Span<int> M(int[] arg) { return arg; }
+            }
+            """;
+
+        var comp = CreateCompilationWithSpan(source);
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+
+        var arg = tree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().Single().Expression;
+        Assert.Equal("arg", arg!.ToString());
+
+        var argType = model.GetTypeInfo(arg);
+        Assert.Equal("System.Int32[]", argType.Type.ToTestDisplayString());
+        Assert.Equal("System.Span<System.Int32>", argType.ConvertedType.ToTestDisplayString());
+
+        var argConv = model.GetConversion(arg);
+        Assert.True(argConv.IsSpan);
+        Assert.True(argConv.IsImplicit);
+        Assert.False(argConv.IsUserDefined);
+        Assert.False(argConv.IsIdentity);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_UnrelatedElementType(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                System.Span<string> M(int[] arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,41): error CS0029: Cannot implicitly convert type 'int[]' to 'System.Span<string>'
+            //     System.Span<string> M(int[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("int[]", "System.Span<string>").WithLocation(3, 41));
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis(LanguageVersion langVersion)
+    {
+        var source = """
+            #nullable enable
+            class C
+            {
+                System.Span<string> M1(string[] arg) => arg;
+                System.Span<string> M2(string?[] arg) => arg;
+                System.Span<string?> M3(string[] arg) => arg;
```

Comment:
>System.Span<string?> M3(string[] arg) => arg; [](http://example.com/codeflow?start=16&length=45)

Should the conversion from `string[]` to `Span<string?>` generate a warning? Otherwise, the following will be allowed without warning:
```csharp
string[] x = new[] { "" };
Span<string?> y = M3(x);
y[0] = null;
x[0].ToString();
```

Reply:
I'm not sure. But arrays also don't generate a warning here:

```cs
string[] x = new[] { "" };
string?[] y = x; // no warning
y[0] = null;
x[0].ToString(); // NullReferenceException
```

Reply:
I ended up changing nullability analysis a bit and now there's a warning.